silkrpc/ethbackend/backend_interface.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+   Copyright 2021 The Silkrpc Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#ifndef SILKRPC_ETHBACKEND_BACKEND_INTERFACE_HPP_
+#define SILKRPC_ETHBACKEND_BACKEND_INTERFACE_HPP_
+
+#include <string>
+
+#include <asio/io_context.hpp>
+#include <asio/use_awaitable.hpp>
+#include <evmc/evmc.hpp>
+#include <silkrpc/types/execution_payload.hpp>
+
+namespace silkrpc::ethbackend {
+
+class BackEndInterface {
```

Comment:
It still will be necessary, there are two main problems: constructor of backend cannot be void and we would need to find a way to create garbage parameters, which would look ugly. second: we would then need to wrap the mock class which i suppose will inherit BackEnd and not BackEndInterface into a unique_ptr, which since it is not abstract will execute the normal methods instead of the mock ones

Reply:
nevermid, i have an idea to get around these